src/System.Private.CoreLib/shared/System/Globalization/HijriCalendar.Win32.cs

```diff
@@ -55,10 +55,10 @@ private static int GetAdvanceHijriDate()
                 }
 
                 int hijriAdvance = 0;
-                string str = value.ToString();
+                string? str = value.ToString();
                 if (string.Compare(str, 0, HijriAdvanceRegKeyEntry, 0, HijriAdvanceRegKeyEntry.Length, StringComparison.OrdinalIgnoreCase) == 0)
                 {
-                    if (str.Length == HijriAdvanceRegKeyEntry.Length)
+                    if (str!.Length == HijriAdvanceRegKeyEntry.Length)
```

Comment:
I had to use `!` because the compiler doesn't know `string.Compare` would compare, `null` with constant `HijriAdvanceRegKeyEntry` which will never be true if str is null.

Reply:
BTW, this also wouldn't have worked if you would have written `if (str == HijriAdvanceRegKeyEntry)`.

Reply:
I know, but I didn't write it, it was like that, so didn't want to change it -- but maybe it wouldn't hurt to change it.

Reply:
Ah right.